forum/qa-include/pages/question-view.php

```diff
@@ -449,11 +449,21 @@ function qa_page_q_answer_view($question, $answer, $isselected, $usershtml, $for
 			$clicksuffix=' onclick="return qa_answer_click('.qa_js($answerid).', '.qa_js($question['postid']).', this);"';
 
 			if ($question['aselectable'] && !$answer['hidden'] && !$answer['queued']) {
-				if ($isselected)
+				if ($isselected){
 					$a_view['unselect_tags']='title="'.qa_lang_html('question/unselect_popup').'" name="'.$prefix.'dounselect"'.$clicksuffix;
-				else
-					$a_view['select_tags']='title="'.qa_lang_html('question/select_popup').'" name="'.$prefix.'doselect"'.$clicksuffix;
-			}
+				}else{
+					$query = 'SELECT * FROM qa_posts WHERE postid='.$answerid;
+					$answerAuthorFromDatabase = qa_db_query_sub($query);
+					foreach($answerAuthorFromDatabase as $answerAuthor2)
+					{
+						$answerAuthor = $answerAuthor2['userid'];
+					}
+					if($answerAuthor === $userid){

```

Comment:
Jak zapytanie by nie zwróciło jakimś cudem nic i pętla się nie wykona to zmiennej $answerAuthor nie będzie